components/calendar/src/types.rs

```diff
@@ -25,25 +25,150 @@ pub struct DateFields<'a> {
     /// The era code as defined by CLDR, represented as ASCII bytes.
     ///
     /// If set, [`Self::era_year`] must also be set.
+    ///
+    /// # Examples
+    ///
+    /// To set the era field, convert an ASCII string to bytes:
```

Comment:
I'd rather say "UTF-8 code units" than "ASCII bytes", because `str::as_bytes()` guarantees the former but not the latter. With this ASCII phrasing I'm seeing callers validating their strings to be ASCII before passing them, which is unnecessary.

We also use UTF-8 terminology everywhere else, even if APIs only work on ASCII.

Reply:
Good point. We should ideally still indicate that only ASCII makes sense but we should say it in a way that doesn't get people prevalidating.